helix.go

```diff
@@ -267,7 +269,12 @@ func (c *Client) newRequest(method, path string, data interface{}, hasJSONBody b
 }
 
 func (c *Client) newStandardRequest(method, url string, data interface{}) (*http.Request, error) {
-	req, err := http.NewRequest(method, url, nil)
+	ctx := c.opts.Context
+	if ctx == nil {
+		ctx = context.Background()
+	}
```

Comment:
I'm thinking it may be advantageous to move this logic into the [`NewClient()`](https://github.com/nicklaw5/helix/blob/1e2b88eeb7bc6d1a916185a739460c16ca465f6e/helix.go#L111-L129) function instead. That way we don't have to include the logic in both for this method and the `newJSONRequest` method. Thoughts?

Reply:
I like that idea better. I made a change to add `NewClientWithContext` to follow along with the standard library, and then made `NewClient` call `NewClientWithContext(context.Background(), ...)`. There's a lot of changes because I needed to update all the tests that create a client manually to have a context or `http.Do` would error.
Another note, I changed the tests I added to not used the newMockClient, but to instead call `NewClient` and `NewClientWithContext` since we want to test that those methods handle clients properly.